MarkBind/markbind
# 2416 fix gitignore not generated bug

**What is the purpose of this pull request?**
Fixes #2416 




**Overview of changes:**
Rename `.gitignore` to `gitignore` in template/default and rename `gitignore` to `.gitignore` when user run `markbind-cli init`

**Anything you'd like to highlight/discuss:**
I have tested it locally, but I think we can only know if this method actually works until we publish the package on npm, not sure if we are allowed to publish a draft package? 

**Testing instructions:**
Make the changes in this PR and `npm link` an empty folder to the local `markbind-cli`
Ensure that local markbind-cli is linked to local @markbind-core
eg, 
In the `packages/cli` directory, 
Run `npm ls --link` , should see `@markbind/core@5.3.0 -> ./packages/core`

In the empty folder, run `npx markbind-cli init`, a `.gitignore` file will be created



**Proposed commit message: (wrap lines at 72 characters)**
Change .gitignore of the default template to gitignore

The default template has a .gitignore file

.gitignore will be ignored by npm and is not able to be published.
The file will not be created in the users' project.

Let's rename it to gitignore so that it can be published and rename 
it back to .gitignore in the users' project.



---

**Checklist:** :ballot_box_with_check:






---

**Reviewer checklist:**

Indicate the [SEMVER](https://semver.org/) impact of the PR:

At the end of the review, please label the PR with the appropriate label: `r.Major`, `r.Minor`, `r.Patch`.

Breaking change release note preparation (if applicable):
- To be included in the release note for any feature that is made obsolete/breaking

> Give a brief explanation note about:
>   - what was the old feature that was made obsolete
>   - any replacement feature (if any), and
>   - how the author should modify his website to migrate from the old feature to the replacement feature (if possible).
